pithos/pithos.py

```diff
@@ -1250,6 +1251,14 @@ def destroy_ui_loop(self):
             GLib.source_remove(self.ui_loop_timer_id)
             self.ui_loop_timer_id = 0
 
+    def search_activate_handler(self,station):
+        row = self.stations_popover.listbox.get_row_at_y(0)
+        if not row:
+            return False
+        self.active_station_changed(self.stations_popover.listbox,row)
+        self.stations_popover.hide()
+        self.stations_popover.search.set_text('')
```

Comment:
Maybe the hide logic should be moved to `StationsPopover.py` as it is for the row activation. No strong opinion.

Reply:
Updated it to just call `on_row_activated`.